pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -382,6 +453,99 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 	return nil
 }
 
+func getContainerWithName(pod *v1.Pod, name string) *v1.Container {
+	for idx, container := range pod.Spec.Containers {
+		if strings.Compare(container.Name, name) == 0 {
+			return &pod.Spec.Containers[idx]
+		}
+	}
+	return nil
+}
+
+// Add an environment variable with the given value to the environments
+// variables of the given container, potentially replacing an env var that already exists
+// with this name
+func addOrReplaceEnv(container *v1.Container, envKey string, envValue string) {
+	if container.Env == nil {
+		container.Env = []v1.EnvVar{}
+	}
+
+	for i, envVar := range container.Env {
+		if envVar.Name == envKey {
+			container.Env[i].Value = envValue
+			return
+		}
+	}
+
+	container.Env = append(container.Env, v1.EnvVar{
+		Name:  envKey,
+		Value: envValue,
+	})
+}
+
+func (mi *StorageInitializerInjector) createModelContainer(image string, modelPath string) *v1.Container {
```

Comment:
In KServe 0.11.1 [this PR](https://github.com/kserve/kserve/pull/3060) introduced `ClusterStorageContainer` which allows configuring custom storage container based on supported storage uri prefixes. For example you can customize a storage container using `oci://` prefix and in this way we no longer need to hard code the container spec here. We can also add an annotation to indicate injecting as sidecar or init container.

Reply:
Let me dive into the ClusterStorageContainer feature, we might also add it with such a CR. As you say, we probably need to extend the mechanism to allow side-car based injection.

Reply:
@yuzisun I've added a section to the [design document](https://docs.google.com/document/d/1Bs4fnP8rhPMaoPoLSYVvuRq-z9vkGPQ0rKbmfH4I7js/edit#bookmark=id.qb0sqvo6n1bq), that describes what changes would be needed in `ClusterStorageContainer` CRD to support model cars.

Please let me know if this is a potential way to go, I would then create an issue for these extensions and might be able to work on it, too.